### PR TITLE
detect-replace: Convert unittests to FAIL/PASS API -v4

### PR DESCRIPTION
**Pull request by Rai0Catodic0**
Ticket: [4054](https://redmine.openinfosecfoundation.org/issues/4054)

- [ :white_check_mark: ] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [ :white_check_mark: ] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR:  #8049 

Describe changes:
-  Started implementing changes on int DetectReplaceLongPatternMatchTest
- Fix code style errors

### CODE REVIEW

File: src/detect-replace.c [hunk 1]
```diff
@@ -276,15 +276,11 @@ int DetectReplaceLongPatternMatchTest(uint8_t *raw_eth_pkt, uint16_t pktsize,
     DecodeEthernet(&th_v, &dtv, p, GET_PKT_DATA(p), pktsize);
```
Comment (Rai0Catodic0) [inline src/detect-replace.c:277]: a few if statements in the test have printfs and debug prints. I'm not sure how I should deal with then, any suggestions

File: src/detect-replace.c [hunk 2]
```diff
@@ -276,15 +276,11 @@ int DetectReplaceLongPatternMatchTest(uint8_t *raw_eth_pkt, uint16_t pktsize,
...
-        goto end;
-    }
+    FAIL_IF_NULL(de_ctx);
     de_ctx->flags |= DE_QUIET;
 
     de_ctx->sig_list = SigInit(de_ctx, sig);
-    if (de_ctx->sig_list == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(DetectEngineAppendSig(de_ctx, sig));
```
Comment (Rai0Catodic0) [inline src/detect-replace.c:283]: Is it right to use the DetectEngineAppendSig function here?

File: src/detect-replace.c [hunk 3]
```diff
@@ -276,15 +276,11 @@ int DetectReplaceLongPatternMatchTest(uint8_t *raw_eth_pkt, uint16_t pktsize,
...
     de_ctx->sig_list->next = NULL;
 
     if (de_ctx->sig_list->sm_lists_tail[DETECT_SM_LIST_PMATCH]->type == DETECT_CONTENT) {
```
Comment (Rai0Catodic0) [inline src/detect-replace.c:286]: this if is an example of what I asked earlier.

**Comment by Rai0Catodic0** [comment]
Hi Reviewer. there are a few parts of the code that I wasn't able to write a comment on because I didn't make any changes on them and I didn't know that GitHub wouldn't allow me to comment on them, sorry about that. 
I asked if a should use the DetectEngineAppendSig function and that brought me to another question. 
at the end of the test function, there are a few functions that I believe are related to cleaning up resources after the test. 
with the addition of the FAIL/PASS macros, what should be the approach to that part of the code? 

Thanks for the review and sorry for the big text :)

File: src/detect-replace.c (hunk 1 above)
Comment (jufajardini) [reply src/detect-replace.c:277]: It's ok to leave them there, thanks for asking :)

File: src/detect-replace.c (hunk 2 above)
Comment (jufajardini) [reply src/detect-replace.c:283]: From my understanding, it would be, yes.

File: src/detect-replace.c (hunk 3 above)
Comment (jufajardini) [reply src/detect-replace.c:286]: I'm rethinking what I had asked, sorry again about that. Let's leave this function untouched. There are many places where the if statements are not meant to mean to be an instant fail, but rather sort of debug what is happening along the way.

I have still answered your other questions, so you know what would be the approach, if you were to proceed with this patch :)

**Review by jufajardini — CHANGES_REQUESTED** [review]
Thanks for this work :)

Thinking again about the test you were converting, I don't think it's a good idea to convert it, after all.

Therefore, let's just remove the changes you've started to make on it, and then, from my end, this contribution would be ready to go.

Thanks for your patience ^^

File: src/detect-replace.c (hunk 2 above)
Comment (victorjulien) [reply src/detect-replace.c:283]: `DetectEngineAppendSig` is preferred, but needs to be used instead of `SigInit`

**Review by victorjulien — CHANGES_REQUESTED** [review]

**Comment by Rai0Catodic0** [comment]
Closing in favour of a newer PR: #8078

**Pull request closed without merging**